### DISCUSSION TITLE
Convert to SVG primitive, from JPEG, PNG

### DISCUSSION THREAD

**Discussion by uasan** [libvips/libvips#2004] [Ideas]
Hello.

We have an interesting task, creating vector photo previews, as described in this article
https://jmperezperez.com/svg-placeholders/

But we do not want to use third-party programs.

Perhaps you will this work useful and interesting and the native function Vips will appear, to create vector primitives from raster images?

Thanks.

**Comment by jcupitt** [comment]
Hello, yes, I've seen that article, it's fun.

It's built on this library:

https://github.com/fogleman/primitive

That would need to be rewritten in C or C++ before we could call it from libvips, sadly.
